jetty-core/jetty-siwe/src/main/java/org/eclipse/jetty/security/siwe/internal/AnyUserLoginService.java

```diff
@@ -0,0 +1,109 @@
+//
+// ========================================================================
+// Copyright (c) 1995 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.security.siwe.internal;
+
+import java.util.function.Function;
+import javax.security.auth.Subject;
+
+import org.eclipse.jetty.security.DefaultIdentityService;
+import org.eclipse.jetty.security.IdentityService;
+import org.eclipse.jetty.security.LoginService;
+import org.eclipse.jetty.security.UserIdentity;
+import org.eclipse.jetty.security.UserPrincipal;
+import org.eclipse.jetty.server.Request;
+import org.eclipse.jetty.server.Session;
+
+public class AnyUserLoginService implements LoginService
```

Comment:
javadoc.

Also, why have we not needed this class before? How is any-user authorization handled by other login services?

Reply:
added javadoc.

> why have we not needed this class before? How is any-user authorization handled by other login services?

We did need it before in OpenId with the `OpenIdLoginService` which is very similar to this. I think they could be combined, but not sure I can really change the `OpenIdLoginService` as it is public class.

Reply:
Then let's not create another specific one. Move it to somewhere common in core.

Reply:
Unlike the `OpenIdLoginService` this one is private so it can be removed/moved at anytime. So I think we should move it to be a public class only once we have another place to use it.